File: Client/src/app/models/KitchenProductDetailsDTO.ts
import { WalmartProductDTO } from "./WalmartProductDTO";

export class KitchenProductDetailsDTO {
  id?: number = undefined;
  name?: string = undefined;
  amount?: number = undefined;
  kitchenUnitType?: number = undefined;
  productId?: number | undefined = undefined;
  walmartProduct?: WalmartProductDTO;
  productSearchItems?: WalmartProductDTO[] = undefined;
}

File: API/ContainerNinja.Contracts/DTO/KitchenProductDetailsDTO.cs

using ContainerNinja.Contracts.Enum;

namespace ContainerNinja.Contracts.DTO
{
    public class KitchenProductDetailsDTO
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public float? Amount { get; set; }

        public KitchenUnitType KitchenUnitType { get; set; }

        public int? ProductId { get; set; }

        public WalmartProductDTO Product { get; set; }

        public IList<WalmartProductDTO> ProductSearchItems { get; set; } = new List<WalmartProductDTO>();
    }
}